pkg/commands/ls.go

```diff
@@ -113,7 +113,14 @@ func mergeStagingComponents(deployed []apis.ComponentMeta, env *types.EnvMeta, i
 					v.Traits[j] = t
 				}
 				aspec.Components[i] = v
+				/* Before any traits is attached to the workload, if ApplicationConfiguration is retrieved through
+				OpenAPI, the .spec.Components[].Traits is nil; if it's resembled by local file, .spec.Components[].Traits
+				is the []v1alpha2.ComponentTrait, 0). Let's set it to be equal before comparing the whole .Spec field.*/
+				if compMeta.AppConfig.Spec.Components[i].Traits == nil && len(v.Traits) == 0 {
```

Comment:
Change to following:

```Go
if  len(compMeta.AppConfig.Spec.Components[i].Traits) == 0 {
  compMeta.AppConfig.Spec.Components[i].Traits = nil
}
if len(v.Traits) == 0 {
  v.Traits = nil
}
```

Reply:
I intended to check both  `compMeta.AppConfig.Spec.Components[i].Traits == nil` and `len(v.Traits) == 0`. If only one is `nil` or `make([]v1alpha2.ComponentTrait, 0`, we don't want to do anything. And they are different which means the status of the application should be `staging`.

is it better?
```
if compMeta.AppConfig.Spec.Components[i].Traits == nil && len(v.Traits) == 0 {
					aspec.Components[i].Traits = nil
				}
```